www/base/src/app/buildslaves/buildslaves.controller.coffee

```diff
@@ -17,9 +17,34 @@ class Buildslaves extends Controller
         buildbotService.all('masters').bind $scope,
             onchild: (master) ->
                 $scope.mastersById[master.masterid] = master
-        buildbotService.all('buildslaves').bind($scope)
+        buildbotService.all('buildslaves').bind $scope,
+            onchild: (buildslave) ->
+                buildslave.some('builds', order: '-started_at', complete: false).bind $scope,
+                dest:buildslave
 
         $scope.settings = bbSettingsService.getSettingsGroup("Slaves")
         $scope.$watch('settings', ->
             bbSettingsService.save()
         , true)
+
+        $scope.slavestatus = {}
+        # Updates the status of a given slave
+        updateSlaveStatus = (slave) ->
+            status = if slave.connected_to.length > 0 then "online" else "not connected"
+            # Looking in our scope fields if there's a match
+            if status == "online"
+                if slave.builds.length > 0
+                    status = "running"
+                else
+                    status = "idle"
+            $scope.slavestatus[slave.name] = status
+
+        # Updates the status of all slaves
+        updateSlavesStatus = (a, b) ->
+            for slave in $scope.buildslaves
+                updateSlaveStatus(slave)
+
+        $scope.$watch('buildslaves', updateSlavesStatus, true)
```

Comment:
I understand the need to narrow watching to only what need to be actually watch, but connected_to is not sufficient because we need to differentiate idle from running